internal/sysmonitor/cpu_linux.go

```diff
@@ -0,0 +1,207 @@
+//go:build linux
+
+package sysmonitor
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"math"
+	"os"
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// linuxProcessSampler samples CPU usage for the current process on Linux
+type linuxProcessSampler struct {
+	fs          FileSystem
+	pid         int
+	lastUTime   float64
+	lastSTime   float64
+	lastSample  time.Time
+	lastPercent float64
+	clockTicks  int64
+}
+
+// newPlatformCPUSampler is the factory entry point used by cpu.go
+func newPlatformCPUSampler(fs FileSystem) (ProcessCPUSampler, error) {
+	pid := os.Getpid()
+	if pid < 0 || pid > math.MaxInt32 {
+		return nil, fmt.Errorf("invalid PID: %d", pid)
+	}
+
+	// We attempt to get clock ticks using the provided FS.
+	// If it fails, we fallback to 100.
+	ticks, err := getClockTicks(fs)
+	if err != nil {
+		ticks = 100
+	}
+
+	return &linuxProcessSampler{
+		fs:         fs,
+		pid:        pid,
+		clockTicks: ticks,
+	}, nil
+}
+
+// Sample returns the CPU usage percentage since the last sample
+func (s *linuxProcessSampler) Sample(deltaTime time.Duration) float64 {
```

Comment:
There is a possible race condition if `Sample` is called concurrently (same in other samplers).